src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -1591,6 +1592,57 @@ public string ToString([StringSyntax(StringSyntaxAttribute.NumericFormat)] strin
             return BigNumber.FormatBigInteger(this, format, NumberFormatInfo.GetInstance(provider));
         }
 
+        private string DebuggerDisplay
+        {
+            get
+            {
+                // For very big numbers, ToString can be too long or even timeout for Visual Studio to display
+                // Display an fast estimation value instead
+
+                // Use ToString for small values
+
+                if (_bits is null || _bits.Length <= 2)
+                {
+                    return ToString();
+                }
+
+                // Estimate the value x as L*2^n, while L is the value of high bits, and n is the length of low bits
+                // Represent L as k*10^i, then x = L*2^n = k*10^(i+n*Log10(2))
+                // Let m=n*Log10(2), the final result would be x = (k*10^(m-[m]))*10^(i+[m])
+
+                ulong highBits = ((ulong)_bits[^1] << kcbitUint) + _bits[^2];
+                double lowBitsCount32 = _bits.Length - 2; // if Length > int.MaxValue/32, counting in bits can cause overflow
```

Comment:
Is this overflow possible, given we limit the length of the BigInteger's `_bits` array to `Array.MaxLength/sizeof(uint32)`? https://source.dot.net/#System.Runtime.Numerics/System/Numerics/BigInteger.cs,624

Reply:
I didn't see the limit. If so, overflow won't be possible.